packages/assets-controllers/src/NftController.test.ts

```diff
@@ -573,15 +657,25 @@ describe('NftController', () => {
 
       const callActionSpy = jest
         .spyOn(messenger, 'call')
+        // 1. `AccountsController:getAccount`
         .mockReturnValueOnce(OWNER_ACCOUNT)
+        // 2. `AssetsContractController:getERC721OwnerOf`
+        .mockResolvedValueOnce(OWNER_ADDRESS)
+        // 3. `AssetsContractController:getERC721TokenURI`
+        .mockResolvedValueOnce('https://testtokenuri.com')
+        // 4. `ApprovalController:addRequest`
         .mockResolvedValueOnce({})
-        .mockReturnValueOnce(OWNER_ACCOUNT);
+        // 5. `AccountsController:getAccount`
+        .mockReturnValueOnce(OWNER_ACCOUNT)
+        // 6. `AssetsContractController:getERC721AssetName`
+        .mockResolvedValueOnce('testERC721Name')
+        // 7. `AssetsContractController:getERC721AssetSymbol`
+        .mockResolvedValueOnce('testERC721Symbol');
 
       await nftController.watchNft(ERC721_NFT, ERC721, 'https://test-dapp.com');
-      // First call is getInternalAccount. Second call is the approval request.
-      expect(callActionSpy).toHaveBeenCalledTimes(3);
+      expect(callActionSpy).toHaveBeenCalledTimes(7);
       expect(callActionSpy).toHaveBeenNthCalledWith(
-        2,
+        4,
         'ApprovalController:addRequest',
```

Comment:
This is the flow of NftController method calls and messenger call invocations triggered by a `watchNft` call on a ERC721 asset.

```md
- `#getAddressOrSelectedAddress`
  1. `AccountsController:getAccount`
- `#validateWatchNft` > `isNftOwner`
  2. `AssetsContractController:getERC721OwnerOf`
- `#getNftInformation` > `#getNftInformationFromTokenURI` > `#getNftURIAndStandard`
  3. `AssetsContractController:getERC721TokenURI`
- `_requestApproval`
  4. `ApprovalController:addRequest`
- `addNft` > `#getAddressOrSelectedAddress`
  5. `AccountsController:getAccount`
- `#addNftContract` > `#getNftContractInformation` > `#getNftContractInformationFromContract`
  6. `AssetsContractController:getERC721AssetName`
  7. `AssetsContractController:getERC721AssetSymbol`
```